src/main/java/luke/LukeUiWindow.java

```diff
@@ -0,0 +1,118 @@
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+        String welcome = """
+                hey, i'm luke, your glorified task manager.
+
+                i can keep a list of your todos, events, and deadlines.
+
+                tell me what you need to keep track of. i'll help you out :)
+                """;
+        dialogContainer.getChildren().add(DialogBox.getLukeDialog(welcome, lukeImage));
+        try {
+            List<String> lines = Storage.loadData();
+            for (String line : lines) {
+                Command command = Parser.parseSavedData(line);
+                Ui.handleCommand(command, true);
+            }
+        } catch (NoSaveDataFoundException e) {
+            handleMissingFile();
+        } catch (IOException e) {
+            String warning = "hmmm... i ran into an issue while setting up. try launching me again.";
+            dialogContainer.getChildren().add(DialogBox.getLukeDialog(warning, lukeImage));
+            System.exit(0);
+        }
+    }
+
+    /**
+     * Injects the Duke instance
+     */
+    public void setLuke(Luke l) {
+        luke = l;
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Luke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
+        String input = userInput.getText();
+        String response = luke.getOutput(input);
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(input, userImage),
+                DialogBox.getLukeDialog(response, lukeImage)
+        );
+        userInput.clear();
+    }
+
+    /**
+     * Provides the user with the choice to create a save file if a save file cannot be found.
+     * Exits if an IOException is thrown or if the user chooses not to make a save file.
+     */
+    public void handleMissingFile() {
+        String createSaveFileQuestion = "i couldn't find a saved task list. "
+                + "you will need to create one to continue using me.\n"
+                + "would you like to create one? (y/n)";
+        dialogContainer.getChildren().add(DialogBox.getLukeDialog(createSaveFileQuestion, lukeImage));
+        // callback??? the setOnAction code block runs once the input (y/n) is typed and the user presses enter.
+        userInput.setOnAction(event -> {
+            String input = userInput.getText().strip().toLowerCase();
+            String response;
+            switch (input) {
+            case "y", "yes" -> {
+                try {
+                    Storage.createSaveFile();
+                    response = "save file created! ok, i'm all ears now. tell me what you need.";
+                } catch (IOException e) {
+                    response = "oof, i couldn't create the file. i'll exit first - try restarting me!";
+                    // System.exit(0);
```

Comment:
Maybe the unused code can be removed instead of commented?

Reply:
Ooh, I missed this one out. Thank you!